theano/tensor/elemwise.py

```diff
@@ -125,6 +125,7 @@ class DimShuffle(Op):
 
     _f16_ok = True
     check_input = False
+    __props__ = ("input_broadcastable", "new_order")
```

Comment:
Yes, i think i somewhat understand this. Will look into this and get back if i have any question regarding this. (Sorry for the delayed response, i was down with flu)